src/js/historical.cpp

```diff
@@ -171,38 +165,16 @@ namespace ccf::js
       return ccf::js::constants::Null;
     }
 
-    ccf::js::Context* jsctx = (ccf::js::Context*)JS_GetContextOpaque(ctx);
+    js::Context& jsctx = *(js::Context*)JS_GetContextOpaque(ctx);
 
-    auto states_array = jsctx->new_array();
+    auto states_array = jsctx.new_array();
     JS_CHECK_EXC(states_array);
     size_t i = 0;
     for (auto& state : states)
     {
-      auto js_state = jsctx->new_obj_class(historical_state_class_id);
-      JS_CHECK_EXC(js_state);
-
-      // Note: The state_ctx object is deleted by js_historical_state_finalizer
-      // which is registered as the finalizer for historical_state_class_id.
-      auto state_ctx = new HistoricalStateContext{
-        state, state->store->create_read_only_tx(), ReadOnlyTxContext{nullptr}};
-      state_ctx->tx_ctx.tx = &state_ctx->tx;
-      JS_SetOpaque(js_state, state_ctx);
-
-      auto transaction_id =
-        jsctx->new_string(state->transaction_id.to_str().c_str());
-      JS_CHECK_EXC(transaction_id);
-      JS_CHECK_SET(js_state.set("transactionId", std::move(transaction_id)));
-
-      auto js_receipt = (*jsctx)(ccf_receipt_to_js(jsctx, state->receipt));
-      JS_CHECK_EXC(js_receipt);
-      JS_CHECK_SET(js_state.set("receipt", std::move(js_receipt)));
-
-      auto kv = jsctx->new_obj_class(kv_read_only_class_id);
-      JS_CHECK_EXC(kv);
-      JS_SetOpaque(kv, &state_ctx->tx_ctx);
-      JS_CHECK_SET(js_state.set("kv", std::move(kv)));
-
-      states_array.set_at_index(i++, std::move(js_state));
+      auto js_state = ccf::js::create_historical_state_object(jsctx, state);
+      // TODO: Wrapper for this?
```

Comment:
set_at_index(i++, std::move(js_state)), but js_state will need to wrapped if it's not